radixdlt-engine/src/main/java/com/radixdlt/atom/TxLowLevelBuilder.java

```diff
@@ -159,37 +147,38 @@ public void instruction(REInstruction.REMicroOp op, byte[] data) {
     }
   }
 
-  public TxLowLevelBuilder message(byte[] bytes) throws MessageTooLongException {
-    if (bytes.length > 255) {
-      throw new MessageTooLongException(bytes.length);
+  TxLowLevelBuilder message(byte[] bytes, int limit) throws MessageTooLongException {
```

Comment:
This also looks like antipattern, should just include `limit` in TxLowLevelBuilder constructor

Reply:
It is a required coupling though as it related to what's allowed to be built. It shouldn't be the caller who makes the decision on the limit. And if the caller is the decider then why can't the caller make the check on its own.